miqa/core/rest/frame.py

```diff
@@ -125,13 +125,21 @@ def create(self, request, *args, **kwargs):
             scan.save()
         elif 'scan' in serializer.data:
             scan = Scan.objects.get(id=serializer.data['scan'])
+            if not scan:
+                raise APIException(
+                    'Could not create new Frame.'
```

Comment:
This should probably have status=404

Reply:
Also, we should probably make the message more informative, like "Scan not found"

Reply:
Changed in [1d6a5b5](https://github.com/OpenImaging/miqa/pull/634/commits/1d6a5b53988507eca80fd0184717ac0fbd3534e4)